doc/SharedComponents/MongoDB/MongoDB.md

```diff
@@ -1,10 +1,112 @@
-# MongoDB
 
-### Purpose
+# Purpose
 
+Our current SDN ApplicationLayer architecture uses Elasticsearch to store and query JSON objects. While Elasticsearch provides powerful capabilities for search and analytics, it is not well-suited for use cases involving frequent updates to JSON documents, especially when those updates involve nested fields(like the one in the new NMDA datastore concepts).
 
-### Design 
+In Elasticsearch, any update, even to a single nested field results in the entire document being reindexed. This leads to:
 
-(Picture describing how we use it)
+- High I/O overhead, as documents must be deleted and reinserted  
+- Segment bloat, where frequent updates generate many small segments that degrade performance  
+- Increased heap usage and garbage collection pressure, especially under write-heavy workloads  
+- Reduced indexing throughput and query responsiveness over time
 
-### ParameterDesign
+These characteristics make Elasticsearch inefficient for workloads where data updation is critical.
+
+To overcome these challenges, we are introducing **MongoDB** as an alternate. MongoDB is a document-oriented database for JSON objects that offers native support for partial updates, allowing us to efficiently modify specific fields within nested JSON structures without rewriting the entire document. 
+
+This change aligns well with our evolving requirements for applications (like **LILW**, **DDM**) with **NMDA datastore** concepts where the information models have nested JSON objects that undergo frequent updates (e.g., adding an entry to the `ControlConstruct` array inside the `NetworkControllerDomain`), where JSON objects are modified frequently and consistency of structure must be maintained.
+
+MongoDB will thus serve as a **highly performant, update-friendly datastore**, while Elasticsearch can continue to serve its strengths in **search and analytics**.
+
+![your-UML-diagram-name](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/openBackhaul/ApplicationPattern/refs/heads/PrathibaJee/MongoDB/doc/SharedComponents/MongoDB/diagram/MongoDBDesign.iuml)
+
```

Comment:
maybe you could add a placeholder box in both "Apps with NMDA DS" and "Applications" to indicate that there can be more apps using MongoDB or ElasticSearch and the ones given here are just examples